### PR TITLE
Replace custom LinkButton with react-router's Link

### PR DESCRIPTION
**Pull request by jseppi**
The API of our custom `LinkButton`, which really just added some button-specific classes and a `role` attribute, was unhelpfully masking use of the underlying `Link` component from `react-router`.  This PR removes our custom component in favor using `Link` directly in all instances.

It also mitigates #1051 by adding `rel="noopener norefer"` to the View Website button in the site header.

Supersedes #1052

### CODE REVIEW

File: test/frontend/components/siteList/siteList.test.js [hunk 1]
```diff
@@ -47,21 +45,19 @@ describe('<SiteList />', () => {
     });
 
     it('always renders 2 `add new site` button', () => {
-      expect(wrapper.find(LinkButton)).to.have.length(2);
+      expect(wrapper.find('Link[to="/sites/new"]')).to.have.length(2);
```
Comment (jseppi) [inline test/frontend/components/siteList/siteList.test.js]: Here's the relevant change to this test

File: frontend/components/site/pagesHeader.js [hunk 2]
```diff
@@ -1,51 +1,41 @@
...
-  }
-}
+    </div>
+    <div className="usa-width-one-third">
+      <Link
+        role="button"
+        className="usa-button usa-button-big pull-right icon icon-view icon-white"
+        alt="View this website"
+        target="_blank"
+        rel="noopener noreferrer"
```
Comment (jseppi) [inline frontend/components/site/pagesHeader.js]: This is the change to mitigate #1051

File: frontend/components/site/pagesHeader.js [hunk 3]
```diff
@@ -1,51 +1,41 @@
...
-          />
-          <div className="header-title">
-            <h1>{repository}</h1>
-            <p>{title}</p>
-          </div>
-        </div>
-        <div className="usa-width-one-third">
-          <LinkButton {...buttonConfigs} />
-        </div>
+const PagesHeader = ({ repository, title, viewLink }) => (
```
Comment (jmhooper) [inline frontend/components/site/pagesHeader.js]: Nice change :+1:

Comment (jseppi) [inline frontend/components/site/pagesHeader.js]: eslint told me to do it :)

**Pull request merged**